# HG changeset patch
# Parent 09088959502f6d05f265787d44e011b04d0add29
# User Ben Kelly <ben@wanderview.com>
Hack QuotaManager not to require client ID at origin init.  Do not take this off maple.


diff --git a/dom/quota/QuotaManager.cpp b/dom/quota/QuotaManager.cpp
--- a/dom/quota/QuotaManager.cpp
+++ b/dom/quota/QuotaManager.cpp
@@ -2012,24 +2012,21 @@ QuotaManager::InitializeOrigin(Persisten
 
     if (!isDirectory) {
       NS_WARNING("Unknown file found!");
       return NS_ERROR_UNEXPECTED;
     }
 
     Client::Type clientType;
     rv = Client::TypeFromText(leafName, clientType);
-    if (NS_FAILED(rv)) {
-      NS_WARNING("Unknown directory found!");
-      return NS_ERROR_UNEXPECTED;
+    if (NS_SUCCEEDED(rv)) {
+      rv = mClients[clientType]->InitOrigin(aPersistenceType, aGroup, aOrigin,
+                                            usageInfo);
+      NS_ENSURE_SUCCESS(rv, rv);
     }
-
-    rv = mClients[clientType]->InitOrigin(aPersistenceType, aGroup, aOrigin,
-                                          usageInfo);
-    NS_ENSURE_SUCCESS(rv, rv);
   }
 
   if (trackQuota) {
     uint64_t quotaMaxBytes;
     uint64_t totalUsageBytes = usageInfo->TotalUsage();
 
     if (IsTreatedAsTemporary(aPersistenceType, aOrigin, aIsApp)) {
       // Temporary storage has no limit for origin usage (there's a group and
